src/com/nutrons/framework/producers/Serial.java

```diff
@@ -0,0 +1,57 @@
+package com.nutrons.framework.producers;
+
+import static com.nutrons.framework.util.FlowOperators.toFlow;
+
+import edu.wpi.first.wpilibj.SerialPort;
+
+import io.reactivex.Flowable;
+
+/**
+ * A wrapper around WPI's SerialE class which provides Flowables for data sent to the roboRIO over serial
+ */
+public class Serial {
+    private final SerialPort serial;
+    private final int bufferSize;
+    private final int packetLength;
+    private final Flowable<byte[]> dataStream;
+    private final char terminationCharacter;
+
+    public Serial(int bufferSize, int packetLength, char terminationCharacter) {
+        this(9600, SerialPort.Port.kUSB, bufferSize, packetLength, '\n');
+    }
+
+    public Serial(SerialPort.Port port, int bufferSize, int packetLength, char terminationCharacter) {
+        this(9600, port, bufferSize, packetLength, terminationCharacter);
+    }
+
+    /**
+     * Create Serial streams from a WPI "Serial"
+     * bufferSize represents how many bytes to cache unread before clearing buffer, packetLength represents the length of each read from the buffer
+     */
+    public Serial(int baudrate, SerialPort.Port port, int bufferSize, int packetLength, char terminationCharacter) {
+        this.serial = new SerialPort(baudrate, port);
+        this.bufferSize = bufferSize;
+        this.packetLength = packetLength;
+        this.terminationCharacter = terminationCharacter;
+
+        if (terminationCharacter == '\n') {
+            serial.enableTermination();
```

Comment:
Check the source to see if `enableTermination()` is implemented as a convenience wrapper around `enableTermination(char terminationChar)`.  If they're the same, simplify this.